crates/core/src/db/models.rs

```diff
@@ -135,3 +142,39 @@ pub struct Storefront<'a> {
     /// The file URL for this store's log
     pub logo_url: Cow<'a, str>,
 }
+
+/// Join record for the RPC getListings query
+#[derive(Debug, Clone, Queryable)]
+pub struct RpcGetListingsJoin {
+    // Table `listings`
+    /// Listing address
+    pub address: String,
+    /// Listing end time
+    pub ends_at: Option<NaiveDateTime>,
+    /// Listing created time
+    pub created_at: NaiveDateTime,
+    /// Listing ended flag
+    pub ended: bool,
+    /// Listing last bid price
+    pub last_bid: Option<i64>,
+    /// Listing price floor
+    pub price_floor: Option<i64>,
+    /// Listing bid count
+    pub total_uncancelled_bids: Option<i32>,
+    /// Listing instant-sale price
+    pub instant_sale_price: Option<i64>,
+
+    // Table `storefronts`
+    /// Storefront subdomain
+    pub subdomain: String,
+    /// Storefront title
+    pub store_title: String,
```

Comment:
Can these fields be nested in a storefront object. storefront.subdomain storefront.title